django/utils/decorators.py

```diff
@@ -45,8 +45,18 @@ def _dec(obj):
         else:
             func = obj
 
+        # helper function to manage the iterable decorator case
+        def decorate(function):
+            if hasattr(decorator, "__iter__"):
+                # at this point we don't know if decorator respects
+                # the sequence protocol so we generate a tuple out of it
+                for dec in reversed(tuple(decorator)):
```

Comment:
Using the list comprehension on the tuple is a good idea, but I'd use reversed because I was thinking it was more readable (i.e. make a tuple out of it and then reverse it).

Decorator is mutated to a tuple because I'm only assuming that it has the `__iter__` attribute whilst don't knowing if `__getitem__` is here or not. I don't know how much useful (99% of time it is probably only a list or a tuple) this precaution is, yet in the documentation I'd write about iterables not sequences.

If you think it is fitting I can just assume it is a sequence since it is a lot of less hassle.